common/pom.xml

```diff
@@ -54,5 +61,112 @@
+        <version>1.9.1</version>
+        <executions>
+          <execution>
+            <id>add-source</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>add-source</goal>
+            </goals>
+            <configuration>
+              <sources>
+                <source>${collection.src.dir}</source>
+              </sources>
+            </configuration>
+          </execution>
+          <execution>
+            <id>add-test-source</id>
+            <phase>generate-test-sources</phase>
+            <goals>
+              <goal>add-test-source</goal>
+            </goals>
+            <configuration>
+              <sources>
+                <source>${collection.testsrc.dir}</source>
+              </sources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+
+      <!-- Generate the primitive collections from the template files. -->
+      <plugin>
+        <groupId>org.codehaus.gmaven</groupId>
+        <artifactId>groovy-maven-plugin</artifactId>
+        <version>2.0</version>
+        <dependencies>
+          <dependency>
+            <groupId>org.codehaus.groovy</groupId>
+            <artifactId>groovy-all</artifactId>
+            <version>2.4.3</version>
+          </dependency>
+          <dependency>
+            <groupId>ant</groupId>
+            <artifactId>ant-optional</artifactId>
+            <version>1.5.3-1</version>
+          </dependency>
+        </dependencies>
+        <executions>
+          <execution>
+            <id>generate-collections</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>execute</goal>
+            </goals>
+            <configuration>
+              <source>
+                String[] templateDirs = ["${collection.template.dir}",
+                                         "${collection.template.test.dir}"]
+                String[] outputDirs = ["${collection.src.dir}",
+                                       "${collection.testsrc.dir}"]
+
+                templateDirs.eachWithIndex { templateDir, i ->
+                  convertSources templateDir, outputDirs[i]
+                }
+
+                void convertSources(String templateDir, String outputDir) {
+                  String[] keyPrimitives = ["int", "char"]
+                  String[] keyObjects = ["Integer", "Character"];
+
+                  keyPrimitives.eachWithIndex { keyPrimitive, i ->
+                    convertTemplates templateDir, outputDir, keyPrimitive, keyObjects[i]
+                  }
+                }
+
+                void convertTemplates(String templateDir,
+                                      String outputDir,
+                                      String keyPrimitive,
+                                      String keyObject) {
+                  def keyName = keyPrimitive.capitalize()
+                  def replaceFrom = "^(.*)[[a-z]+\\.]*K([A-Za-z_]+).template"
```

Comment:
@Scottmitch so what (if any) action are you suggesting here. This works, I'm tempted to leave it as-is. It actually worked before as well, and was simpler ... should I revert?